src/lib.rs

```diff
@@ -160,11 +160,47 @@ mod std_support;
 mod test_util;
 #[cfg(feature = "u128")]
 mod u128_support;
-#[cfg(feature = "v3")]
+#[cfg(all(
+    feature = "v3",
+    any(
+        not(target_arch = "wasm32"),
+        all(
+            target_arch = "wasm32",
+            any(
+                all(feature = "stdweb", not(feature = "wasm-bindgen")),
+                all(feature = "wasm-bindgen", not(feature = "stdweb"))
+            )
+        )
+    )
+))]
 mod v3;
-#[cfg(feature = "v4")]
+#[cfg(all(
+    feature = "v4",
+    any(
+        not(target_arch = "wasm32"),
+        all(
+            target_arch = "wasm32",
+            any(
+                all(feature = "stdweb", not(feature = "wasm-bindgen")),
+                all(feature = "wasm-bindgen", not(feature = "stdweb"))
+            )
+        )
+    )
+))]
 mod v4;
-#[cfg(feature = "v5")]
+#[cfg(all(
+    feature = "v5",
+    any(
+        not(target_arch = "wasm32"),
+        all(
+            target_arch = "wasm32",
+            any(
+                all(feature = "stdweb", not(feature = "wasm-bindgen")),
```

Comment:
do we need both features to be mutually exclusive? what will happen if this condition doesn't exist and both features exist?

Reply:
According to [this](https://github.com/rust-random/rand/blob/master/src/rngs/os.rs#L1004-L1006) and [this](https://github.com/rust-random/rand/blob/master/src/rngs/os.rs#L1107-L1110) I assume that the compilation will fail, because of two `imp` modules. And we should probably add `not(target_os = "emscripten")` as well, to mimick `rand` behavior.

Reply:
Sounds good to me :+1:

Reply:
Not 100% sure if `not(target_os = "emscripten")` should be added, because [at the top level](https://github.com/rust-random/rand/blob/master/src/lib.rs#L169-L170) only `wasm32` & `stdweb` | `wasm-bindgen` is used.

Reply:
My assumption was wrong.

`cargo build --target wasm32-unknown-unknown --features "stdweb wasm-bindgen"` works and `stdweb` is picked.

Rand contains ...

```
#[cfg(all(target_arch = "wasm32",
          not(target_os = "emscripten"),
feature = "stdweb"))]
```

... which works when `stdweb` is enabled. And it also contains ...

```
#[cfg(all(target_arch = "wasm32",
          not(target_os = "emscripten"),
          not(feature = "stdweb"),
feature = "wasm-bindgen"))]
```

... which utilizes `wasm-bindgen` only if `stdweb` is disabled.

* It means that `stdweb` & `wasm-bindgen` does not need to be mutually exclusive. `rand` can handle this
* Still not sure if `not(target_os = "emscripten")` should be added

Reply:
I don't think RNGs are supported on emscripten which means if we don't bother about it, then compilation will fail and if we include it in the feature gate, it will compile to nothing.